AQCU-1052 first pass fix for extremes report

mhines-usgs opened the pull request:
error seen, when primary and related upchain number of values mismatch an internal error occurred
now, it merges the two datasets into one dataframe and assigns NAs to values missing in one or the other.

mhines-usgs committed:
first pass fix for extremes not rendering when primary and related upchain number of values mismatch.

dpattermann-usgs commented on R/extremes-data.R:
```diff
@@ -210,7 +212,27 @@ createDataRows <-
+        
+        if (!isEmptyOrBlank(x$relatedPrimary)) {
+          if (nrow(x$relatedPrimary) > nrow(x$points) ||
+              nrow(x$points) > nrow(x$relatedPrimary)) {
```
Could these two if statements be simplified to nrow(x$related{Primary or Upchain}) != nrow(x$points) ?

mhines-usgs replied:
is that a thing?! i'm actually writing a whole separate function now to reduce the duplicity of these two sets of statements as well and will try that..

dpattermann-usgs replied:
Ha, I just mean replace each of those statements, it won't just be one statement.

dpattermann-usgs replied:
Like (if(nrow(x$relatedPrimary) != nrow(x$points){ blah }
       if(nrow(x$relatedUpchain) != nrow(x$points)){ blah }
Sorry for the confusion, I don't have any secret R magic to get rid of the duplication off the top of my head.

mhines-usgs committed:
reducing duplicity of work into a new function

mhines-usgs committed:
using other example fabricated by Laura in AQCU-1052, realized that I also need to override relatedValue if relatedSet isn't null and the dataframes are different sizes

mhines-usgs committed:
two tests for merging data for extremes

mhines-usgs committed:
new test file for merging data in extremes report

mhines-usgs committed:
adding a footnote related to changes in AQCU-1052; if uneven data exists, a footnote will appear

mhines-usgs committed:
fixing a few tests i broke changing format of data around for AQCU-1052

zmoore-usgs reviewed (changes requested):
The approach looks good to me, though I'd suggest instead of giving each row the footnote string perhaps "footnote" should just hold TRUE or FALSE and then instead of including the if/else in the for loop in extremesTable I think you could just do return(list(toRet=toRet, footnote=any(dataRows$footnote))

Then in the RMD file add the actual footnote text if footnote=true, similar to how the other notes are being added about the '*' characters

mhines-usgs committed:
last of broken tests

mhines-usgs commented:
thanks @zmoore-usgs now that I got all the tests working again, I'll give that a try.

mhines-usgs committed:
merge conflict with PTs versionstring changes

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	inst/extremes/extremes.Rmd

zmoore-usgs commented:
I took this ticket from @mhines-usgs after she left and made a few tweaks. New PR: https://github.com/USGS-R/repgen/pull/637
